OOPHomework14January/src/Store.java

```diff
@@ -0,0 +1,292 @@
+     * @param temperature  - the optimal temperature to preserve the product
+     * @param id           - the id of the product
+     */
+    public void createAnimalProduct(String productName, float price, String validityDate,
+                             int weight, int stock, int temperature, int id) {
+        // check the ID, we don't want to add duplicates
+        if (duplicateID(id)) {
+            //create a new AnimalProduct
+            arrayAnimalProduct.add(new AnimalProduct(productName, price, validityDate, weight, stock, temperature, id));
+        } else {
+            System.out.println("The product " + productName +
+                    " was not added to the Store!!!");
+            System.out.println("The id given to " + productName +
+                    " is already given to another product.");
+        }
+    }
+
+    /**
+     * method to add a new product to the Store
+     *
+     * @param productName  - the name of the product
+     * @param price        - the price of the product
+     * @param validityDate - the validity date of the product
+     * @param weight       - the weight of the product
+     * @param stock        - the stock of the product
+     * @param vitamins     - the list of vitamins
+     * @param id           - the id of the product
+     */
+    public void createVegetalProduct(String productName, float price, String validityDate,
+                              int weight, int stock, int id, ArrayList<String> vitamins) {
+        // check the ID, we don't want to add duplicates
+        if (duplicateID(id)) {
+            //create a new AnimalProduct
+            arrayVegetalProduct.add(new VegetalProduct(productName, price, validityDate, weight, stock, id, vitamins));
+        } else {
+            System.out.println("The product " + productName +
+                    " was not added to the Store!!!");
+            System.out.println("The id given to " + productName +
+                    " is already given to another product.");
+        }
+    }
+
+
+    /**
+     * method to sell a product
+     *
+     * @param id       - product's id that will be sold
+     * @param quantity - the quantity we want to sell
+     */
+    public boolean sellAnimalProduct(int id, int quantity) {
+        boolean found = false;
+        for (AnimalProduct animalProduct : arrayAnimalProduct) {
+            // if in the array of animal products exists a product with this id, and we have the appropriate quantity
+            if ((animalProduct.getId() == id) && (animalProduct.getStock() >= quantity)) {
+                // the product will be added in an array of orders
+                arrayOrders.add(new Order(localDate(), id, quantity));
+                // set the new stock of the product after selling
+                animalProduct.setStock(animalProduct.getStock() - quantity);
+                found = true;
+            } else if ((animalProduct.getId() == id) && (animalProduct.getStock() < quantity)) {
```

Comment:
you're doing to ifs, each with 2 conditions, you could do 2 ifs, each with 1 condition:
```
if(animalProduct.getId() == id) {
    if(animalProduct.getStock>=quantity) {
        // prod exists, with enough quantity
    } else {
        // prod exists, without enough quantity
    }
}
```